frescobaldi_app/musicview/widget.py

```diff
@@ -100,9 +95,7 @@ def openDocument(self, doc):
         """Open a qpageview.Document instance."""
         self._links = None
         self._highlightRange = None
-        document = doc.document()
-        if document:
-            self._links = pointandclick.links(document)
+        self._links = pointandclick.links(doc)
```

Comment:
Point and click data don't refresh after an engrave job finishes and this change is guilty. (First reported [here](https://github.com/frescobaldi/frescobaldi/pull/1780#issuecomment-2298557403), but the problem is more general than originally thought.)

On master/PyQt5 the `doc.document()` is a new instance of `popplerqt5.Poppler.Document` after every build, while `document` remains the same instance of `qpageview.poppler.PopplerDocument`.

The caching mechanism in `musicview.pointandclick.links()` relies on the fact that a new instance is passed whenever the document is engraved. Since now the same instance is passed every time, links cached after the first engraving are reused over and over, disregarding changes of the pdf.

Reply:
I don't have time to look at the code right this second, but a couple thoughts:
* Some of the fix will probably have to happen in qpageview rather than here.
* Frescobaldi shouldn't be messing with backend types like `popplerqt5.Poppler.Document` at all -- though I do note the irony of complaining about leaky abstractions in an editor for LilyPond, a language made of them.
* This section needs better commenting at the very least. Lines like `document = doc.document()` without so much as an indication of the types involved are not as self-explanatory as some seem to think.